### PR TITLE
Export refactor

### LINKED ISSUE #50: Allow downloading of the entire code, sections of code, and chapters of code, in EPUB, Word, and PDF formats
There are two promising potential methods of providing EPUB files: [ePubExport](http://sourceforge.net/projects/epubexport/) and "[EPub](http://www.phpclasses.org/package/6115-PHP-Create-ebook-in-EPUB-format-for-ex-Apple-iPad.html)". The former is a MediaWiki extension, the latter a PHP class. Elmer Masters took the "EPub" class and improved upon it for [Free Law Reporter](https://github.com/emasters/Free-Law-Reporter), specifically found [here](https://github.com/emasters/Free-Law-Reporter/blob/master/includes/EPub.php).

### PR DESCRIPTION
**Pull request by krues8dr**
It only took 18 months, but this looks like it works pretty good now.

This radically changes how all exports & downloads are handled, moving the logic into a plugin-based system.  This introduces the EventManager class for plugin events, called via the `trigger` method.  This adds a lot of new export types – including HTML, Word, PDF, and EPUB – to resolve #50 as well.

### CODE REVIEW

**Comment by waldoj** [comment]
Good Lord. This will take a while to review. :)

File: includes/class.EventManager.inc.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		{
+			unset($this->events[$event][$listener]);
+		}
+		else
+		{
+			unset($this->events[$event]);
+		}
+	}
+
+	public function has($name)
```
Comment (waldoj) [inline includes/class.EventManager.inc.php]: I am impressed by naming a function `has()`. :)

**Comment by waldoj** [comment]
I've been wanting to write a plugin, specific to Virginia—now I can do it. :)

**Comment by krues8dr** [comment]
Ack, I accidentally merged this.  Still needs some review.

**Comment by krues8dr** [comment]
And git still doesn't have a good way of rolling back a merge with history that overlaps other commits.  Sooooo.

**Comment by waldoj** [comment]
Eh, it'll all be reviewed properly, once I get #42 done. I read through all of your commits, but that's not quite the same as actually _testing_ it. :)

**Comment by krues8dr** [comment]
https://www.youtube.com/watch?v=Hk_F0aEVC90

**Pull request merged**